src/main/java/org/jetbrains/bsp/bazel/server/ParsingUtils.java

```diff
@@ -0,0 +1,125 @@
+package org.jetbrains.bsp.bazel.server;
+
+import ch.epfl.scala.bsp4j.Diagnostic;
+import ch.epfl.scala.bsp4j.DiagnosticSeverity;
+import ch.epfl.scala.bsp4j.Position;
+import ch.epfl.scala.bsp4j.Range;
+import com.google.common.base.Splitter;
+import com.google.common.io.Files;
+import com.google.devtools.build.lib.buildeventstream.BuildEventStreamProtos.Progress;
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.jetbrains.bsp.bazel.common.Constants;
+
+public class ParsingUtils {
+
+  private static final Logger LOGGER = LogManager.getLogger(ParsingUtils.class);
+
+  private static final String ERROR = "ERROR";
+  private static final String ERROR_PREAMBLE = ERROR + ": ";
+  private static final String AT_PATH = " at /";
+
+  public static URI parseUri(String uri) {
+    try {
+      return new URI(uri);
+    } catch (URISyntaxException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  // TODO FOR REVIEW: naming suggestions? what exactly is the thing that it returns?
+  public static List<String> parseClasspathFromFile(URI protoPathUri) {
+    try {
+      return Files.readLines(new File(protoPathUri), StandardCharsets.UTF_8).stream()
+          .map(line -> Splitter.on("\"").splitToList(line))
+          .peek(
+              parts -> {
+                if (parts.size() != 3) {
+                  // TODO FOR REVIEW: verify the slight change of semantics
+                  // TODO is it OK to throw an exception at the beginning if any line is corrupted?
```

Comment:
If this by any chance is not fine we probably should handle this better actually. I never actually played around with this logic, but I'm not sure its a good idea for us to just break everything upon this breaking.
The refactor is indeed correct but this should probably just filter out those broken dependencies and signal the error.
Probably this out of the scope of this PR though.

Reply:
Perhaps we could then to create an issue out of this thread?

Reply:
Good idea